src/main/java/com/minecolonies/coremod/util/InventoryFunctions.java

```diff
@@ -134,8 +139,13 @@ private static boolean matchInInventory(
      * @param tester    the function to use for testing slots
      * @return true if it found a stack
      */
-    public static boolean matchFirstInInventory(final IInventory inventory, @NotNull final BiPredicate<Integer, ItemStack> tester)
+    public static boolean matchFirstInProvider(final ICapabilityProvider inventory, @NotNull final BiPredicate<Integer, ItemStack> tester)
     {
-        return matchFirstInInventory(inventory, inv -> slot -> stack -> tester.test(slot, stack));
+        return matchFirstInProvider(inventory, inv -> slot -> stack -> tester.test(slot, stack));
+    }
+
+    @FunctionalInterface
+    public interface IMatchActionResult {
+        void accept(ICapabilityProvider provider, int slotIndex);
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Document this public method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AUndocumentedApi)